module/zfs/dmu_traverse.c

```diff
@@ -41,6 +41,7 @@
 
 int32_t zfs_pd_bytes_max = 50 * 1024 * 1024;	/* 50MB */
 int32_t send_holes_without_birth_time = 1;
+int32_t zfs_traverse_indirect_prefetch_limit = 10;
```

Comment:
Considering this goes about reduction from 1024, I am thinking whether some bigger value like 16 or 32 could have sense to saturate wider pools.  Remember those are async reads, that may not even get to the media due to ZFS I/O scheduler.

Reply:
As I have told, you've already reduced the value ~100 times. Some loyalty to worst case single-threaded workload would be nice.

Reply:
Makes sense. Would change default to 32.